File: app/src/main/java/com/example/despensacx/TiendaAdapter.java
package com.example.despensacx;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.chip.Chip;

import java.util.ArrayList;
import java.util.List;

public class TiendaAdapter extends RecyclerView.Adapter<TiendaAdapter.TiendaViewHolder> {

    public interface OnTiendaActionListener {
        void onEditar(TiendaEntity tienda);
        void onEliminar(TiendaEntity tienda);
    }

    private List<TiendaEntity> tiendas = new ArrayList<>();
    private final OnTiendaActionListener listener;

    public TiendaAdapter(OnTiendaActionListener listener) {
        this.listener = listener;
    }

    public void setTiendas(List<TiendaEntity> tiendas) {
        this.tiendas = tiendas;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public TiendaViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_tienda, parent, false);
        return new TiendaViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull TiendaViewHolder holder, int position) {
        holder.bind(tiendas.get(position), listener);
    }

    @Override
    public int getItemCount() {
        return tiendas.size();
    }

    static class TiendaViewHolder extends RecyclerView.ViewHolder {
        TextView tvNombre;
        View viewColorChip;
        ImageButton btnEditar, btnEliminar;

        public TiendaViewHolder(@NonNull View itemView) {
            super(itemView);
            tvNombre = itemView.findViewById(R.id.tvNombreTienda);
            viewColorChip = itemView.findViewById(R.id.viewColorChip);
            btnEditar = itemView.findViewById(R.id.btnEditarTienda);
            btnEliminar = itemView.findViewById(R.id.btnEliminarTienda);
        }

        public void bind(TiendaEntity tienda, OnTiendaActionListener listener) {
            tvNombre.setText(tienda.getNombre());
            try {
                viewColorChip.setBackgroundColor(Color.parseColor(tienda.getColor()));
            } catch (Exception e) {
                viewColorChip.setBackgroundColor(Color.GRAY);
            }

            btnEditar.setOnClickListener(v -> listener.onEditar(tienda));
            btnEliminar.setOnClickListener(v -> listener.onEliminar(tienda));
        }
    }
}

File: app/src/main/kotlin/com/example/despensacx/adapter/TiendaAdapter.kt
package com.example.despensacx.adapter

import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.despensacx.R
import com.example.despensacx.data.TiendaEntity

class TiendaAdapter(private val listener: OnTiendaActionListener) :
    RecyclerView.Adapter<TiendaAdapter.TiendaViewHolder>() {

    interface OnTiendaActionListener {
        fun onEditar(tienda: TiendaEntity)
        fun onEliminar(tienda: TiendaEntity)
    }

    private var tiendas: List<TiendaEntity> = emptyList()

    fun setTiendas(tiendas: List<TiendaEntity>) {
        this.tiendas = tiendas
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TiendaViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_tienda, parent, false)
        return TiendaViewHolder(view)
    }

    override fun onBindViewHolder(holder: TiendaViewHolder, position: Int) {
        holder.bind(tiendas[position], listener)
    }

    override fun getItemCount(): Int = tiendas.size

    class TiendaViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvNombre: TextView = itemView.findViewById(R.id.tvNombreTienda)
        private val viewColorChip: View = itemView.findViewById(R.id.viewColorChip)
        private val btnEditar: ImageButton = itemView.findViewById(R.id.btnEditarTienda)
        private val btnEliminar: ImageButton = itemView.findViewById(R.id.btnEliminarTienda)

        fun bind(tienda: TiendaEntity, listener: OnTiendaActionListener) {
            tvNombre.text = tienda.nombre
            try {
                viewColorChip.setBackgroundColor(Color.parseColor(tienda.color))
            } catch (e: Exception) {
                viewColorChip.setBackgroundColor(Color.GRAY)
            }

            btnEditar.setOnClickListener { listener.onEditar(tienda) }
            btnEliminar.setOnClickListener { listener.onEliminar(tienda) }
        }
    }
}
